### PR TITLE
Don't crash the generator if api_names_out.yaml is absent

### LINKED ISSUE #694: Problems with api_names_out.yaml
Hey Guys, 

I've experienced some issues while generating the files for the maps/booking API. 
The error was regarding the file `api_names_out.yaml`, in which was removed from the gem assembly on the commit: 5ce622ccd200a67e749456b33bf95251811595a3 , leading to the following error:
```
@rails4/gems/google-api-client-0.23.0/lib/google/apis/generator/annotator.rb:47:in `read': No such file or directory @ rb_sysopen - @rails4/gems/google-api-client-0.23.0/api_names_out.yaml (Errno::ENOENT)
```
To solve this problem, I had to download and move the file manually to the Gem directory, afterwards the files were generated successfully.

I'm wondering if you guys could put this file back in the gem final directory or if you could use some sort of CDN to read this file when necessary.

### PR DESCRIPTION
**Pull request by dazuma**
Fixes #694

### CODE REVIEW

File: lib/google/apis/generator/annotator.rb [hunk 1]
```diff
@@ -42,13 +42,13 @@ class Names
         include NameHelpers
 
         def initialize(names_out_file_path = nil, names_file_path = nil)
-          if names_out_file_path
+          if names_out_file_path && File.file? names_out_file_path
```
Comment (blowmage) [inline lib/google/apis/generator/annotator.rb]: Needs params when chaining conditionals.

```ruby
if names_out_file_path && File.file?(names_out_file_path)
```

File: lib/google/apis/generator/annotator.rb [hunk 2]
```diff
@@ -42,13 +42,13 @@ class Names
...
             logger.info { sprintf('Loading API names from %s', names_out_file_path) }
             @names = YAML.load(File.read(names_out_file_path)) || {}
           else
             @names = {}
           end
-          if names_file_path
+          if names_file_path && File.file? names_file_path
```
Comment (blowmage) [inline lib/google/apis/generator/annotator.rb]: Needs params when chaining conditionals.

```ruby
if names_file_path && File.file?(names_file_path)
```

File: lib/google/apis/generator/annotator.rb (hunk 1 above)
Comment (dazuma) [reply lib/google/apis/generator/annotator.rb]: Crap, I keep forgetting. Thanks!

File: lib/google/apis/generator/annotator.rb (hunk 2 above)
Comment (dazuma) [reply lib/google/apis/generator/annotator.rb]: Fixed.

**Pull request merged**